components/channel-picker.js

```diff
@@ -11,10 +11,20 @@ export default class ChannelPicker extends React.Component {
     const channelsP = slackFetch('https://slack.com/api/channels.list', {
       exclude_archived: 1,
       token: props.auth.token.access_token,
-    })
+    });
+
+    const groupsP = slackFetch('https://slack.com/api/groups.list', {
+      exclude_archived: 1,
+      token: props.auth.token.access_token,
+    });
 
-    channelsP.then(res => {
-      this.setState({channels: res.channels.filter(c => c.is_member)})
+    // Fetch all channels
+    channelsP.then(channelRes => {
+        // Then fetch private groups
+        groupsP.then(groupRes => {
```

Comment:
Let's use `allChannelsP = Promise.all([groupsP, channelsP]).then([groupsRes, channelsRes] => [...channelRes.channels.filter(c => c.is_member), ...groupRes.groups])`, rather than dereferencing both nestedly.

This focuses on promises as eventual values, and less on promises as an async control flow mechanism.